_templates/layout.html

```diff
@@ -0,0 +1,284 @@
+    {%- endif %}
+  {%- endif %}
+  {%- endblock %}
+
+  {%- block linktags %}
+    {%- if hasdoc('about') %}
+    <link rel="author" title="{{ _('About these documents') }}" href="{{ pathto('about') }}" />
+    {%- endif %}
+    {%- if hasdoc('genindex') %}
+    <link rel="index" title="{{ _('Index') }}" href="{{ pathto('genindex') }}" />
+    {%- endif %}
+    {%- if hasdoc('search') %}
+    <link rel="search" title="{{ _('Search') }}" href="{{ pathto('search') }}" />
+    {%- endif %}
+    {%- if hasdoc('copyright') %}
+    <link rel="copyright" title="{{ _('Copyright') }}" href="{{ pathto('copyright') }}" />
+    {%- endif %}
+    {%- if next %}
+    <link rel="next" title="{{ next.title|striptags|e }}" href="{{ next.link|e }}" />
+    {%- endif %}
+    {%- if prev %}
+    <link rel="prev" title="{{ prev.title|striptags|e }}" href="{{ prev.link|e }}" />
+    {%- endif %}
+  {%- endblock %}
+  {%- block extrahead %} {% endblock %}
+</head>
+
+<body class="wy-body-for-nav">
+
+  {%- block extrabody %} {% endblock %}
+  <div class="wy-grid-for-nav">
+    {#- SIDE NAV, TOGGLES ON MOBILE #}
+    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
+      <div class="wy-side-scroll">
+        <div class="wy-side-nav-search" {% if theme_style_nav_header_background %} style="background: {{theme_style_nav_header_background}}" {% endif %}>
+          {%- block sidebartitle %}
+
+          {%- if logo and theme_logo_only %}
+            <a href="{{ pathto(master_doc) }}">
+          {%- else %}
+            <a href="{{ pathto(master_doc) }}" class="icon icon-home"> {{ project }}
+          {%- endif %}
+
+          {%- if logo %}
+            {#- Not strictly valid HTML, but it's the only way to display/scale
+                it properly, without weird scripting or heaps of work
+            #}
+            <img src="{{ pathto('_static/' + logo, 1) }}" class="logo" alt="{{ _('Logo') }}"/>
+          {%- endif %}
+          </a>
+
+          {%- if theme_display_version %}
+            {%- set nav_version = version %}
+            {%- if READTHEDOCS and current_version %}
+              {%- set nav_version = current_version %}
+            {%- endif %}
+            {%- if nav_version %}
+              <div class="version">
+                {{ nav_version }}
+              </div>
+            {%- endif %}
+          {%- endif %}
+
+          {%- include "searchbox.html" %}
+
+          {%- endblock %}
+        </div>
+
+        {%- block navigation %}
+        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="{{ _('Main') }}">
+          {%- block menu %}
+            {%- set toctree = toctree(maxdepth=theme_navigation_depth|int,
+                                      collapse=theme_collapse_navigation|tobool,
+                                      includehidden=theme_includehidden|tobool,
+                                      titles_only=theme_titles_only|tobool) %}
+            {%- if toctree %}
+                <ul>
```

Comment:
Shouldn't this list be generated automatically from the toctree rather than being hard-coded?

Reply:
Ah yes, I am not super happy with that either but that felt like the best option to what I was aiming for.
So there were a couple of things which led to this decision:

1. We have a mix of toctree and headings. 
Before only either the toctree or the headings were shown in the side-bar, but we want to have both in the side-bar:
`            {%- if toctree %}
              {{ toctree }}
            {%- else %}
              <!-- Local TOC -->
              <div class="local-toc">{{ toc }}</div>
            {%- endif %}` 

2. I thought also it looked nicer not to have the first heading "_Teaching and learning material by the Energy, Climate, and Environment Program (ECE)_" in the navigation bar since it felt redundant.
If I implemented the header with `.. raw:: html` to avoid this behaviour,  the _Table of Content_ was not build correctly.
3. Also I thought the page won't change that much - but now thinking about it there will be a lot of new headlines beeing implemented...

Maybe I missed something and there is a more elegant way to do it if one want to incorporate both toctree and headings automatically? I guess I would have to update the iiasa.css file for that? I will check it out!